streams/src/main/java/org/apache/kafka/streams/state/internals/WrappingStoreProvider.java

```diff
@@ -28,20 +29,25 @@
 public class WrappingStoreProvider implements StateStoreProvider {
 
     private final List<StreamThreadStateStoreProvider> storeProviders;
-    private final boolean includeStaleStores;
+    private StoreQueryParams storeQueryParams;
 
     WrappingStoreProvider(final List<StreamThreadStateStoreProvider> storeProviders,
-                          final boolean includeStaleStores) {
+                          final StoreQueryParams storeQueryParams) {
         this.storeProviders = storeProviders;
-        this.includeStaleStores = includeStaleStores;
+        this.storeQueryParams = storeQueryParams;
+    }
+
+    //visible for testing
+    public void setStoreQueryParams(final StoreQueryParams storeQueryParams) {
+        this.storeQueryParams = storeQueryParams;
     }
```

Comment:
Why do we need this setter? Why can't we pass in `StoreQueryParams` in the constructor in the tests?

Reply:
There were four different tests in this class, with different storeName or queryable store type for each test. So, I would have to create 4 different instances of wrappingStoreProvider in the #before (one for each test) if I don't add a setter. I thought this to easier. I can surely do that as well. LMK.

Reply:
Yes, it would be cleaner to create a new instance for each test -- if you test `WrappingStoreProvider` for a different use case, it makes total sense to create a new object. We should only pock holes into the API if it is absolutely necessary to keep clean API abstractions.